internal/controller/k8s/k8scluster/cluster.go

```diff
@@ -145,12 +147,30 @@ func (c *externalCluster) Observe(ctx context.Context, mg resource.Managed) (man
 		ResourceExists:          true,
 		ResourceUpToDate:        k8scluster.IsK8sClusterUpToDate(cr, observed),
 		ResourceLateInitialized: !cmp.Equal(current, &cr.Spec.ForProvider),
-		ConnectionDetails: managed.ConnectionDetails{
-			"kubeconfig": []byte(kubeconfig),
-		},
+		ConnectionDetails:       createKubernetesConnectionDetails(c, kubeconfig, mg),
```

Comment:
Since we can't get the kubeconfig until the cluster is in active state, I suggest to change a bit the logic and move getting the config and populating the fields to when the cluster goes to active state. 

	if strings.EqualFold(cr.Status.AtProvider.State, k8s.ACTIVE) {
		if kubeconfig, _, err = c.service.GetKubeConfig(ctx, cr.Status.AtProvider.ClusterID); err != nil {
			c.log.Info(fmt.Sprintf("failed to get connection details. error: %v", err))
		}
		mo.ConnectionDetails = createKubernetesConnectionDetails(c, kubeconfig, mg)
	}

Reply:
Great idea. I also had this thought but wanted to keep the api consistent. Should i Still add this?

Reply:
yes, please. the backend will just throw 422 until the cluster is in active anyway

Reply:
might be better to change the condition to !strings.EqualFold(cr.Status.AtProvider.State, k8s.DEPLOYING). Will do some testing and fix the tests

Reply:
thank you :) giving this PR the last finishing touches. 

Thanks for the easy process and the fast responses.